packages/cli-kit/src/private/node/ui/components/SelectInput.tsx

```diff
@@ -232,20 +250,30 @@ function SelectInputInner<T>(
       </Box>
     )
   } else {
+    if (hasLimit) useStdout().stdout.write(ansiEscapes.clearTerminal)
```

Comment:
Remember to put `useStdout` outside of `if` conditions otherwise React will complain. I believe if you use the `eslint` plugin in vscode it will tell you that this is an error.